src/main/java/Duke/activityAndUtility/Deadline.java

```diff
@@ -0,0 +1,82 @@
+package Duke.activityAndUtility;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * The {@code Deadline} class represents a deadline activity, implementing the {@link Activity} interface. It encapsulates
+ * the status, name, and due date and time of a task. This class is designed to manage deadlines, allowing for them to be
+ * printed, named, and marked as complete or incomplete.
+ */
+public class Deadline implements Activity {
+    List<String> act;
```

Comment:
I think a better use of OOP is to abstract out the List management to another TaskManager class.

Deadline, event and todo should be just classes to represent a deadline, .. object.